File: 2024/c++/day1part1.cpp
#include <bits/stdc++.h>
#include <iostream>
#include <cmath>
using namespace std;

vector<string> split(const string &s, char delimiter) {
    vector<string> tokens;
    string token;
    stringstream ss(s); 
    
    while (getline(ss, token, delimiter)) {
        tokens.push_back(token);
    }
    
    return tokens;
}


int main() {
    ifstream file("../Input/day1.1.txt");
    if (!file.is_open()) {
        cerr << "Error opening the file!";
        return 1;
    }
    string line;
    vector<int> array1;
    vector<int> array2;
    while (getline(file, line)) {
        array1.push_back(stoi(split(line, ' ').at(0)));
        array2.push_back(stoi(split(line, ' ').at(3)));
        cout << line << endl;
    }
    stable_sort(array1.begin(), array1.end());
    stable_sort(array2.begin(), array2.end());

    int sum = 0;
    for (int i = 0; i < array1.size(); i++) {
        sum += abs(array1.at(i) - array2.at(i));
    }
    file.close();
    cout << sum << endl;
    return 0;
}

File: 2024/Java/Day1Part1.java
import java.io.*;
import java.nio.file.*;
import java.nio.file.Path;
import java.util.*;

class Day1Part1 {
    public static void main(String[] args) throws IOException{
        ArrayList<Integer> array1 = new ArrayList<>();
        ArrayList<Integer> array2 = new ArrayList<>();
        try (BufferedReader reader = Files.newBufferedReader(Path.of("2024/Input/day1.1.txt"))) {
            String line;
            while ((line = reader.readLine()) != null) {
                line.trim();
                array1.add(Integer.parseInt(line.split(" ")[0]));
                array2.add(Integer.parseInt(line.split(" ")[3]));
                //System.out.println(line.split(" ")[0] + " " + line.split(" ")[3]);
            }
        }
        Collections.sort(array1);
        Collections.sort(array2);

        int sum = 0;
        for (int i = 0; i < array1.size(); i++) {
            sum += Math.abs(array1.get(i) - array2.get(i));
        }
        System.out.println(sum);
    }
}